pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -167,6 +170,18 @@ private static void validateGroupByClause(PinotQuery pinotQuery)
     }
   }
 
+  private static boolean containsAggregateAndColsInSelectClause(PinotQuery pinotQuery) {
+    int aggregateExprCount = 0;
+    int identifierExprCount = 0;
+    for (Expression expr : pinotQuery.getSelectList()) {
+      if (isAggregateExpression(expr)) {
+        aggregateExprCount++;
+      } else if (expressionContainsColumn(expr)) {
```

Comment:
I feel we can simplify the checks. Without group-by clause, the select expressions should be either all aggregations or no aggregations (I don't think we need to support mixing literals and aggregations)

Reply:
Makes sense to avoid the heavy function call, will make the change.